### DISCUSSION TITLE
I need help getting started with my FPGA

### DISCUSSION THREAD

**Discussion by mrmejer** [NUS-CG3207/labs#13] [Wrangling Vivado] [answered]
I went through the entire [Getting Started](https://github.com/NUS-CG3207/lab-manuals/blob/4b14f6d6df9ea91f8b1c5293c303f1fcfeeb5846/getting_started.pdf) manual, followed all the steps very carefully, made sure the schematic designs look right, but after programming the FPGA, I could not see any effect (the leds were not responding to my switches).

To make sure that I have everything set up properly, I tried programming my FPGA with the binary from [simple_count.bit](https://github.com/NUS-CG3207/lab-skeletons/blob/main/lab1/simple_count_Verilog/simple_count_Verilog.runs/impl_1/simple_count.bit) If I understand it correctly, it should display 4 bits constantly counting up unless BTNC is pressed. And again, nothing is happening on my FPGA. I tried reruning synthesis, implementation and programming but nothing helps. Is that the desired effect? How can I be sure that the FPGA has been programmed? Do I have to somehow start the FPGA after programming it or should it just start working on its own right after being programmed?

**Comment by kxing28** [comment]
Lets see if the FPGA itself works. When you set the blue plastic piece in "QSPI" like this picture:
![image](https://github.com/user-attachments/assets/a5239a90-7242-4f31-8546-a37590a0645b)

and turn it on, do the 7 segment displays say "PASS" and then start some animation? If so, then the board components work.

If you power off, switch the plastic piece to the middle portion, under "JTAG" like this picture:
![image](https://github.com/user-attachments/assets/e3bfc44a-d954-462b-97f2-0a6fadd2d16a)
and then power on + attempt to program device, does it work now?


Also note that the original repo has a .bit file that worked on my FPGA right out of the box (NEXYS4 not ddr version)

**Reply by mrmejer** [reply]
When I turned it on, it did not show PASS but started the animation right away. I then switched to JTAG like you said and programmed with the .bin from repo. The DONE diode turned green, but nothing else happened. 

Vivado says the FPGA is programmed:
![image](https://github.com/user-attachments/assets/96eaf513-8955-4279-9b47-b926593ac7e5)

I also have a DDR version of the nexys4, maybe that is the problem?

**Reply by NeilBaner** [reply, marked as answer]
Make sure you are using the correct constraints file; the Nexys 4 and Nexys 4 DDR use different files. The FPGA part number itself is the same between the two boards, but the constraints aren't.

**Reply by mrmejer** [reply]
Yes!!!! That worked, thank you so much.
